eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -73,19 +79,56 @@ function Update-AutorestConfigFile() {
                     $configline = "input-file:"  + "$inputfile"
                 }
             }
-            $fileContent = Get-Content -Path $autorestFilePath
+
             $fileContent[$startNum - 1] += ([Environment]::NewLine + $configline)
             $fileContent | Set-Content $autorestFilePath
             if ( !$? ) {
                 Write-Error "Failed to update autorest.md. exit code: $?"
                 exit 1
             }
-        }  
+        }
+        
+        # update autorest.md with configuration
+        if ( $autorestConfigYaml -ne "") {
+            Write-Host "Update autorest.md with configuration."
+            $range = ($autorestConfigYaml | Select-String -Pattern '```').LineNumber
+            if ( $range.count -gt 1) {
+                $startNum = $range[0];
+                $lines = $range[1] - $range[0] - 1
+                $autorestConfigYaml = ($autorestConfigYaml | Select -Skip $startNum | Select -First $lines) |Out-String
+            }
+
+            Install-Module -Name powershell-yaml -Force -Verbose -Scope CurrentUser
+            Import-Module powershell-yaml
+            $yml = ConvertFrom-YAML $autorestConfigYaml
+
+            $fileContent = Get-Content -Path $autorestFilePath
+            Foreach ( $key in $yml.keys) {
+                if ( ($key -eq "output-folder") -or ($key -eq "require")) {
+                    continue;
+                }
+                $match = ($fileContent | Select-String -Pattern $key).LineNumber
+                if ($match.count -gt 0) {
+                    $fileContent[$match[0] - 1] = $key + ":" + $yml[$key];
+                } else {
+                    $startNum = ($fileContent | Select-String -Pattern '```').LineNumber[0]
+                    $fileContent[$startNum - 1] += ([Environment]::NewLine + $key + ":" + $yml[$key])
+                }
+            }
+
+            $fileContent | Out-File $autorestFilePath
+        }
     } else {
-        Write-Error "autorest.md doesn't exist."
-        exit 1
+        Write-Host "autorest.md does not exist. start to create one."
+        if ( $autorestConfigYaml -ne "") {
```

Comment:
Tip: just `if ($autorestConfigYaml)` is fine. If null or empty, it will evaluate as false anyway.